src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.cpp

```diff
@@ -23,41 +23,38 @@ namespace argos {
       /* Get positional attributes */
       GetNodeAttribute(t_tree, "position", Position);
       GetNodeAttribute(t_tree, "look_at", Target);
-      /* Calculate the Forward vector */
-      Forward = (Target - Position).Normalize();
       /* Was the up vector specified? */
       if(!NodeAttributeExists(t_tree, "up")) {
          /* Calculate the Left vector
             It is located on a plane parallel to XY
             It is perpendicular to the projection of Forward on that plane
          */
-         if(Forward.GetX() != 0 || Forward.GetY() != 0) {
-            CVector2 cLeftXY(Forward.GetX(), Forward.GetY());
+         CVector3 cForward((Target - Position).Normalize());
```

Comment:
This code is wrong. If the difference between `Target` and `Position` is zero, you'll get a `NaN`.

Reply:
I notice now that also the original code had the same bug... Oh well :)

Reply:
I will check this, but I do not think I change the logic of the code only refactored it a bit...